modules/addons/external-dns/main.tf

```diff
@@ -0,0 +1,49 @@
+provider "helm" {
+  kubernetes {
+    host                   = var.k8s_host
+    cluster_ca_certificate = base64decode(var.k8s_cluster_ca_certificate)
+    token                  = var.k8s_client_token
+  }
+}
+
+resource "helm_release" "external_dns" {
+  name             = "external-dns"
+  repository       = "https://charts.bitnami.com/bitnami"
+  chart            = "external-dns"
+  create_namespace = true
+  namespace        = "external-dns"
+  timeout          = 900
+
+  set {
+    name  = "policy"
+    value = "upsert-only"
+  }
+
+  set {
+    name  = "registry"
+    value = "txt"
+  }
+
+  set {
+    name  = "txtOwnerId"
+    value = var.cluster_name
+  }
+
+  # Since we are enabling the Istio Gateway source, it is important to set this filter
+  # so we don't create DNS records for any random or internal hostnames defined in
+  # the TSB IngressGateways.
+  set {
+    name  = "domainFilters"
+    value = "{${var.dns_zone}}"
+  }
+
+  set {
+    name  = "sources"
+    value = "{service,ingress,istio-gateway}"
```

Comment:
+istio-virtualservice ? technically both are supported to be mentioned together.

Reply:
YEs, both are supported but we only need the gateways, which is where the really exposed services are defined. We don't ned VSs ehre.

Reply:
out of curiosity, is there a harm of `istio-virtualservice` be there? i.e. we decided to enforce an exact names under gateway, however if `TID` or `PLZ` will be used - i usually do see `wildcards` on gateway in OSS :) world. so keeping it there would it harm?